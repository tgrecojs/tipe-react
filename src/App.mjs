import React, { Component } from 'react';
import { render } from 'react-dom';
//
import { getData } from './shared/api/index';
import { Provider } from './shared/TipeContext/context';
import { ArticleDetails } from './example'
import { ThemeProvider } from 'emotion-theming'


const theme = {
  color: 'red',
};
const queryStr = `
{
    Webpage(id: "5a8ea44fa61b5500134fcd61") {
     aboutMe
     eggheadLink
     tagline
     pageColor
     techBackground
       }
   }
`;

export const themes = {
  cowboys: {
    // royal blue
    primary: '#003594',
    // dark blue
    secondary: '#003594',
    // silver
    complement: '#869397',
    textColor: 'white',
    fontFamily:  "'Kavivanar', cursive",
    fontSize: '16'
  },
  light: {
    primary: 'white'
  },
  black: {
    primary: 'black'
  }
}

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      data: { },
      theme: themes.cowboys
    }
  }

  componentWillMount() {
    getData(queryStr).then(data => this.setState({ 
      data
    }))

  }


  render() {
    return (
      <Provider value={this.state.data}>
        <ThemeProvider theme={this.state.theme}>
        <ArticleDetails />
        </ThemeProvider>
      </Provider>
    );
  }
}
