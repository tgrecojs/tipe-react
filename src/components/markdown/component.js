import React from 'react';
import styled from 'react-emotion'
import PropTypes from 'prop-types';
const Markdown = require('react-markdown');
const { string, object } = PropTypes

const defaultMdStyles = {
   display: 'flex',
   flexDirection: 'column'
}

const MdComponent = styled(Markdown)(props => ({
    ...defaultMdStyles,
    // trying to find the line between supplying styles / allowing customization
    color: props.theme.textColor,
}))

MdComponent.propTypes = {
  theme: object,
  source: string,
}

const Page = styled('div')(props => ({
  display: !props.theme.flex ? 'flex' : `${props.theme.flex}`,
  flexDirection: !props.theme.flexDirection ? 'column' : `${props.theme.flex}`,
  fontFamily: props.theme.fontFamily,
  backgroundColor:props.theme.primary,
  // trying to find the line between supplying styles / allowing customization
  color: props.theme.textColor,
}))


const RenderMd = ({data}) => <MdComponent source={data}/>


export {
  MdComponent,
  RenderMd,
  Page
}