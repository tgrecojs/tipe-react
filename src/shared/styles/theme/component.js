import React from 'react';
import styled from 'react-emotion/macro'
import { ThemeProvider } from 'emotion-theming'
import renderer from 'react-test-renderer'

const theme = {
  borderRadius: '50%',
  borderColor: '#BF67AD'
}

const Avatar = styled('img')`
  width: 96px;
  height: 96px;
  border-radius: ${props => props.theme.borderRadius};
  border: 1px solid ${props => props.theme.borderColor};
`
const logoUrl = 'https://i.pinimg.com/564x/9b/d7/4f/9bd74fc0e9547d6da161c2c9c6e485e3.jpg';

const ThemeComponent = ({themeConfig = theme}) => 
    <ThemeProvider theme={theme}>
        <Avatar src={logoUrl}></Avatar>
    </ThemeProvider>

export default ThemeComponent
