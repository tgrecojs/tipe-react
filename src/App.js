import React, { Component } from 'react';
import { render } from 'react-dom';
//
import TipeProvider, { TipeConsumer, exampleComponent } from './components/TipeContext/TipeContext.component';
import { getData } from './shared/api/index';
import { Consumer } from './components/TipeContext/consumer.component';
import MarkdownExample from './example'
import { ThemeProvider } from 'emotion-theming'

const Ex = props => {
  console.log('Props from example', props)
  return <h2>I am the example</h2>
}


const App = () => (
      <TipeProvider>
        <TipeConsumer component={MarkdownExample} />
      </TipeProvider>
)

export default App
