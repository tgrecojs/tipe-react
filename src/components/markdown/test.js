import { RenderMd, Page } from './component';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { themes } from '../../App';
import renderer from 'react-test-renderer'

const defaultProps = {
    theme: themes.cowboys
}

const createProps = ({theme, data} = defaultProps) => ({
   ...defaultProps
})



test(`<Page />`, () => {
  expect(
    renderer.create(<Page />).toJSON()
  ).toMatchSnapshot()
})
