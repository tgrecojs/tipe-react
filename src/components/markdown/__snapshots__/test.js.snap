// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> 1`] = `
<div
  className="css-1dmfnyv"
/>
`;
